### PR TITLE
Several Improvements, see comment

### PR DESCRIPTION
**Pull request by KritantaDev**
How running `make setup` looks now:
image

Removed:
* Replaced HBLog with the macros it evaluates to
* Removed simject.h as it was kinda pointless
* Removed theos dependency as it's entirely unnecessary
* Removed the empty control file required by theos

Changes/Additions:
* Dylib directory can now be specified at compile time using the envar DYLIB_DIR (e.g. make setup DYLIB_DIR=<your_dir_here>)
* Arch can be specified with the ARCH envar
* Sysroot can be specified with the SYSROOT envar
* simject.m is now compiled without logos, as it didn't use logos directives anyways.
* The main makefile only uses sudo if it can't already write to the directory
** Makefile also explains to user why it's asking for the root pass
* The main makefile now copies resim to /usr/local/bin/
* Makefile output is a bit prettier (and descriptive)

### CODE REVIEW

**Comment by KritantaDev** [comment]
Didn't merge from upstream before I made the commit with all of these changes, so now I've thrown us into git hell.

**Comment by PoomSmart** [comment]
@KritantaDev Wow, great work there. Could you also resolve conflicts in this PR?

**Comment by angelXwind** [comment]
Changes look good to me! Good work o;

I _will_ say it is sort of… an interesting choice to no longer rely on Theos (at the cost of making the Makefile way more complex l o l), especially given that 99% of jailbreak developers are probably going to be using Theos ;P

That being said, it is definitely true that simject does not _really_ require Theos, so eh, I guess you might as well remove that requirement.

File: resim/Makefile [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
 
-include $(THEOS_MAKE_PATH)/tool.mk
+clean::
+	-@rm -r out/
 
-after-all::
-	@echo "Copying binaries..."
+build : resim.mm
+	@mkdir -p out
+	@echo "$(PREFIX) Compiling Resim for $(shell clang -print-target-triple)"
```
Comment (angelXwind) [inline resim/Makefile:20]:
```suggestion
	@echo "$(PREFIX) Compiling resim for $(shell clang -print-target-triple)"
```

File: simject/Makefile [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+endif
 
-after-all::
-	@echo "Copying binaries..."
+clean::
+	-@rm -r out/
+
+build : simject.m
+	@mkdir -p out
+	@echo "$(PREFIX) Compiling Simject for $(TRIPLE)"
```
Comment (angelXwind) [inline simject/Makefile:25]:
```suggestion
	@echo "$(PREFIX) Compiling simject for $(TRIPLE)"
```

File: Makefile [hunk 3]
```diff
@@ -1,22 +1,44 @@
-ARCHS = x86_64
-DEBUG = 0
+export PREFIX = \033[1;34m>>\033[0m
+export SUB_PREFIX = \033[1;34m>>>\033[0m
```
Comment (angelXwind) [inline Makefile:2]:
```suggestion
export PREFIX = \033[1;36m>>\033[0m
export SUB_PREFIX = \033[1;36m>>>\033[0m
```

File: simject/Makefile [hunk 4]
```diff
@@ -1,14 +1,33 @@
...
+	@echo "$(SUB_PREFIX) Using SDK at $(SYSROOT)"
+	@echo "$(SUB_PREFIX) Target Directory is $(DYLIB_DIR)"
```
Comment (angelXwind) [inline simject/Makefile:27]:
```suggestion
	@echo "$(SUB_PREFIX) Target directory is $(DYLIB_DIR)"
```

File: resim/Makefile [hunk 5]
```diff
@@ -1,15 +1,27 @@
...
+	@echo "$(SUB_PREFIX) Target Directory is $(DYLIB_DIR)"
```
Comment (angelXwind) [inline resim/Makefile:21]:
```suggestion
	@echo "$(SUB_PREFIX) Target directory is $(DYLIB_DIR)"
```

**Comment by KritantaDev** [comment]
I have spent the last 30 minutes trying to fix the merge conflict. Could you try reverting my PR from earlier? My apologies

**Comment by PoomSmart** [comment]
@KritantaDev I guess you can try merging master into your PR and resolve conflicts from there. We then can “squash and merge” afterwards.

**Pull request merged**